src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -0,0 +1,276 @@
+    } else {
+      log.info("Starting calls: " + runnableCalls.map {_.name}.toSeq.sorted.mkString(", "))
+      val futureCallsAndInputs = for {
+        _ <- persistStatus(runnableCalls, Starting)
+        allInputs <- Future.sequence(runnableCalls map fetchLocallyQualifiedInputs)
+      } yield runnableCalls zip allInputs
+      val callsAndInputs = Await.result(futureCallsAndInputs, DatabaseTimeout)
+      log.info("Calls and inputs are " + callsAndInputs)
+      callsAndInputs foreach { case (call, inputs) => startActor(call, inputs)}
+    }
+  }
+
+  // FIXME FIXME FIXME this needs to be refactored but I don't understand it.
+  def fetchLocallyQualifiedInputs(call: Call): Future[Map[String, WdlValue]] = {
+    val workflow = call.parent.map {_.asInstanceOf[Workflow]} getOrElse {
+      throw new WdlExpressionException("Expecting 'call' to have a 'workflow' parent.")
+    }
+
+    object CallInputWdlFunctions extends WdlFunctions {
+      def getFunction(name: String): WdlFunction = {
+        throw new WdlExpressionException("TODO: Some functions may be allowed in this context")
+      }
+    }
+
+    def lookup(identifierString: String): WdlValue = {
+      val namespaces = call.namespace.namespaces filter {_.namespace.contains(identifierString)}
+      namespaces.headOption.getOrElse {
+        val matchedCall = workflow.calls.find {_.name == identifierString}.getOrElse {
+          throw new WdlExpressionException(s"Expecting to find a call with name '$identifierString'")
+        }
+        val futureValue = fetchCallOutputEntries(matchedCall).map { entries =>
+          val callOutputs = entries.map { entry =>
+            val value = entry.wdlValue match {
+              case Some(v) => v
+              case _ => throw new WdlExpressionException(s"Could not evaluate call '${matchedCall.name}', because '${entry.key.name}' is undefined")
+            }
+            entry.key.name -> value
+          }
+          WdlObject(callOutputs.toMap)
+        }
+        Await.result(futureValue, DatabaseTimeout)
+      }
+    }
+
+    log.info("Fetching locally qualified inputs for " + call.fullyQualifiedName)
+    fetchCallInputEntries(call).map { entries =>
+      entries.map { entry =>
+        val value = entry.wdlValue match {
+          case Some(e: WdlExpression) => e.evaluate(lookup, CallInputWdlFunctions).get
+          case Some(v) => v
+          case _ => throw new WdlExpressionException("Unknown error")
+        }
+        entry.key.name -> value
+      }.toMap
+    }
+  }
+
+  private def fetchCallOutputEntries(call: Call) = dataAccess.getOutputs(workflow.id, call)
+  private def fetchCallInputEntries(call: Call) = dataAccess.getInputs(workflow.id, call)
+
+  /**
+   * Load whatever execution statuses are stored for this workflow, regardless of whether this is a workflow being
+   * restarted, or started for the first time.
+   */
+  private def createStore: Future[Map[Call, ExecutionStatus.Value]] = {
+    for {
```

Comment:
I don't understand the for/yield w/ a single clause. Is it getting you something?